src/bifrost.cpp

```diff
@@ -0,0 +1,276 @@
+#include "map_strings.hpp"
+
+// build graph from refs/reads
+ColoredCDBG<> buildGraph (const std::string& infile_1,
+                          const std::string& infile_2,
+                          const bool& is_ref,
+                          const int kmer,
+                          const int threads,
+                          const bool verb,
+                          const bool& write_graph,
+                          const std::string& output_prefix)
+{
+    std::ifstream infile1(infile_1);
+    std::ifstream infile2(infile_2);
+    CCDBG_Build_opt opt;
+
+
+    opt.k = kmer;
+    opt.nb_threads = threads;
+    opt.verbose = verb;
+    opt.prefixFilenameOut = output_prefix;
+
+    std::string filename;
+    if (is_ref && (infile_2 == "NA")) {
+        while (std::getline(infile1, filename))
+        {
+            opt.filename_ref_in.push_back(filename);
+        }
+    } else if (!is_ref && (infile_2 == "NA"))
+    {
+        while (std::getline(infile1, filename))
+        {
+            opt.filename_seq_in.push_back(filename);
+        }
+    } else {
+        while (std::getline(infile1, filename))
+        {
+            opt.filename_ref_in.push_back(filename);
+        }
+        while (std::getline(infile2, filename))
+        {
+            opt.filename_seq_in.push_back(filename);
+        }
+    }
+
+    ColoredCDBG<> ccdbg(opt.k);
+    ccdbg.buildGraph(opt);
+    ccdbg.simplify(opt.deleteIsolated, opt.clipTips, opt.verbose);
+    ccdbg.buildColors(opt);

```

Comment:
Do any of these take a threads option? Or is that already included in the `opt` struct?

Reply:
I believe this is already included in the opt structure, as you pass all other arguments this way.